src/WasmSdk/Tasks/PatchWasmPublishStaticWebAsset.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.NET.Sdk.WebAssembly;
+
+public class PatchWasmPublishStaticWebAsset : Task
+{
+    [Required]
+    public ITaskItem[] StaticWebAssets { get; set; }
+
+    [Output]
+    public ITaskItem[] AssetsToRemove { get; set; }
+
+    public override bool Execute()
+    {
+        var assetDictionary = StaticWebAssets.ToDictionary(a => a.ItemSpec);
+        var assetsToRemove = new List<ITaskItem>();
+
+        foreach (ITaskItem asset in StaticWebAssets)
+        {
+            string assetKey = asset.ItemSpec;
+            assetDictionary[assetKey] = asset;
+            string relatedAsset = asset.GetMetadata("RelatedAsset");
+            if (!string.IsNullOrEmpty(relatedAsset) && !assetDictionary.ContainsKey(relatedAsset))
```

Comment:
Should this happen in second pass over SWA in case where `RelatedAsset` points to a different asset? Is that possible? If not, do we need the dictionary?